app/models/katello/concerns/subscription_aspect_host_extensions.rb

```diff
@@ -5,7 +5,18 @@ module SubscriptionAspectHostExtensions
       include ForemanTasks::Concerns::ActionTriggering
 
       included do
+        SUBSCRIPTION_STATUS_MAP = {
+          :valid => Katello::SubscriptionStatus::FULL_COMPLIANCE,
+          :partial => Katello::SubscriptionStatus::PARTIAL_COMPLIANCE,
+          :invalid => Katello::SubscriptionStatus::INVALID,
+          :unknown => Katello::SubscriptionStatus::UNKNOWN
+        }
+
         has_one :subscription_aspect, :class_name => '::Katello::Host::SubscriptionAspect', :foreign_key => :host_id, :inverse_of => :host, :dependent =>  :destroy
+
+        has_one :subscription_status_object, :class_name => 'Katello::SubscriptionStatus', :foreign_key => 'host_id'
```

Comment:
I follow the convention from foreman:  https://github.com/theforeman/foreman/blob/develop/app/models/concerns/hostext/search.rb#L20

and i do kinda like it as it is really only meant to be used for searching, so having an awkward name makes it obvious when you are using it and shouldn't be.